runelite-client/src/main/java/net/runelite/client/plugins/kittennotifier/KittenNotifierPlugin.java

```diff
@@ -0,0 +1,93 @@
+package net.runelite.client.plugins.kittennotifier;
+import com.google.inject.Provides;
+import net.runelite.api.ChatMessageType;
+import net.runelite.api.NPC;
+import net.runelite.api.events.ChatMessage;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.NpcSpawned;
+import net.runelite.client.Notifier;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.eventbus.Subscribe;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.api.Client;
+import javax.inject.Inject;
+
+@PluginDescriptor
+(
+	name = "Kitten Notifier",
+	description = "Sends a notification when your kitten needs food, attention, or is grown.",
+	tags = {"kitten", "cat", "notifications"}
+)
+public class KittenNotifierPlugin extends Plugin
+{
+	@Inject
+	private Notifier notifier;
+
+	@Inject
+	private KittenNotifierConfig config;
+
+	@Inject
+	private Client client;
+
+	@Provides
+	KittenNotifierConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(KittenNotifierConfig.class);
+	}
+
+	private NPC kitCat;
+	private String kitCatWas;
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (config.kittenNeeds() && event.getType() == ChatMessageType.SERVER)
+		{
+			if (event.getMessage().contentEquals("<col=ef1020>Your kitten is hungry.</col>"))
+			{
+				notifier.notify("Your kitten is hungry.");
+			}
+			if (event.getMessage().contentEquals("<col=ef1020>Your kitten wants attention.</col>"))
+			{
+				notifier.notify("Your kitten wants attention.");
+			}
+			if (event.getMessage().contentEquals("<col=ef1020>Your kitten is very hungry.</col>"))
+			{
+				notifier.notify("Your kitten is very hungry.");
+			}
+			if (event.getMessage().contentEquals("<col=ef1020>Your kitten really wants attention.</col>"))
+			{
+				notifier.notify("Your kitten really wants attention.");
+			}
+		}
+	}
+	@Subscribe
+	public void onGameTick(GameTick event)
+	{
+		if (kitCat != null && kitCat.getName() != null)
```

Comment:
Is there ever a case that the cat's name is actually null?

Flip this if statement so that it early returns if null.

Reply:
In the case that the cat is picked up the object for the NPC still exists with all data cleared, making all attributes in the object null, but not the object itself. I just checked for name because it was easiest.